mephisto/abstractions/architects/local_architect.py

```diff
@@ -145,7 +145,12 @@ def deploy(self) -> str:
         self.running_dir = os.path.join(
             get_mephisto_tmp_dir(), f"local_server_{self.task_run_id}", "server"
         )
-        shutil.copytree(self.server_dir, self.running_dir, symlinks=True)
+
+        # TODO: Not sure that this is correct, but added `dirs_exist_ok=True`
+        #  as this line raised an exception that all these files already exist.
+        #  Maybe we need to remove it and recopy all the time.
+        #  Deployment should cleanup all directories it needs or use previous files
+        shutil.copytree(self.server_dir, self.running_dir, symlinks=True, dirs_exist_ok=True)
```

Comment:
We should only be creating a single directory per task run, and it already existing should be considered an error.

Part of the rationale for retaining the old server was that we could examine the results for this task run in the exact setting they were collected in for debug reasons, even after the server has been shut down. 

If, for instance, you're restarting your database from scratch and using `dirs_exist_ok` you're going to potentially be populating builds that have files from completely different deploys inside.

Underneath, the issue you're adding this to resolve is one that's more fundamental and would need to be addressed at the core - Mephisto doesn't have a good `cleanup` method for `TaskRun`s (or `Requester`s for that matter), and we'd need to update a containerized version of Mephisto to entirely clean up all of the resources.